### PR TITLE
Support OmegaConf in Lightning

### LINKED ISSUE #1880: Support OmegaConf dicts in hparams
## 🚀 Feature
Lightning hparams were recently changed to support only dicts and ArgParse namespaces. This diff (re)introduces compatibility with OmegaConf DictConfig, which is the type of config that Hydra uses.

### Motivation

Using Hydra + Lightning

### PR DESCRIPTION
**Pull request by Darktex**
Lightning hparams were recently changed to support only dicts and ArgParse namespaces. This diff (re)introduces compatibility with OmegaConf DictConfig, which is the type of config that Hydra uses.

## What does this PR do?
Fixes #1880.

### CODE REVIEW

File: pytorch_lightning/trainer/training_io.py [hunk 1]
```diff
@@ -119,6 +119,15 @@
 else:
     HOROVOD_AVAILABLE = True
 
+try:
+    import omegaconf
+except ImportError:
+    OMEGACONF_AVAILABLE = False
+else:
+    OMEGACONF_AVAILABLE = True
+
```
Comment (tullie) [inline pytorch_lightning/trainer/training_io.py:128]: Can you remove these extra whitelines? Should only be 2 between imports and the class def

**Comment by yukw777** [comment]
How does this PR play with this discussion: https://github.com/PyTorchLightning/pytorch-lightning/issues/1871?

**Comment by williamFalcon** [comment]
let’s add support for this now while we work out a good long-term solution.  Thanks for the PR!

**Comment by tullie** [comment]
@Borda Can you explain your comment a bit more? I don't understand the issue

**Comment by Borda** [comment]
> @Borda Can you explain your comment a bit more? I don't understand the issue

https://github.com/PyTorchLightning/pytorch-lightning/blob/9b629637b890a1eb9a0d9011bfa10322e0d89da0/pytorch_lightning/core/lightning.py#L1613-L1617

**Comment by ImMrMa** [comment]
> > @Borda Can you explain your comment a bit more? I don't understand the issue
> 
> https://github.com/PyTorchLightning/pytorch-lightning/blob/9b629637b890a1eb9a0d9011bfa10322e0d89da0/pytorch_lightning/core/lightning.py#L1613-L1617

save the DictConfig to hparms.yaml and load it need to do!

**Comment by Darktex** [comment]
I'm editing this from the WebUI so the experience is... uh... improvable? :D But hopefully this should do the trick

**Comment by Darktex** [comment]
I could add a test for this here https://github.com/PyTorchLightning/pytorch-lightning/blob/master/tests/trainer/test_trainer.py#L29

but it would require to add omegaconf as a dep. What do you guys think?

**Comment by williamFalcon** [comment]
we can add as a test dep but let’s keep it out of the main requirements. we already do this with the loggers as well

**Comment by Borda** [comment]
@Darktex how is it going, mind finish it? it seems that Will was asking for some tests...

**Comment by Darktex** [comment]
Looks like we don't need this anymore after #1896 landed

**Comment by williamFalcon** [comment]
@Darktex i assume this means omegaconf is now supported after we generalized the init?

**Comment by williamFalcon** [comment]
Done in #2025.

**Pull request closed without merging**